python/lsst/pipe/base/cmdLineTask.py

```diff
@@ -464,10 +465,19 @@ def parseAndRun(cls, args=None, config=None, log=None, doReturnResults=False):
             This will typically be a list of `None` unless doReturnResults is `True`;
             see cls.RunnerClass (TaskRunner by default) for more information.
         """
+        if args is None:
+            commandAsStr = " ".join(sys.argv)
+            args = sys.argv[1:]
+        else:
+            commandAsStr = "{}{}".format(lsst.utils.get_caller_name(skip=1), tuple(args))
+
         argumentParser = cls._makeArgumentParser()
         if config is None:
             config = cls.ConfigClass()
         parsedCmd = argumentParser.parse_args(config=config, args=args, log=log, override=cls.applyOverrides)
+        # print this message after parsing the command so the log is fully configured
+        parsedCmd.log.info("Running: {}".format(commandAsStr))
```

Comment:
This is okay but maybe better `parsedCmd.log.info("Running: %s", commandAsStr)` as `log` does the % formatting (and only the % formatting at this moment...)